### PR TITLE
Log filtered redirect chains

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java [hunk 1]
```diff
@@ -66,35 +68,55 @@ protected Set<String> filterResults(Stream<String> urls)
         Stream<String> results = urls;
         if (filterRedirects)
         {
-            Set<String> uniqueUrls = urls.collect(Collectors.toSet());
-            results = uniqueUrls.stream().filter(url -> isNotExistingRedirect(url, uniqueUrls));
+            Set<String> allUrls = urls.collect(Collectors.toSet());
```
Comment (valfirst) [inline vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java:71]: why is it renamed?

File: vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java [hunk 2]
```diff
@@ -66,35 +68,55 @@ protected Set<String> filterResults(Stream<String> urls)
...
+            for (String url : allUrls)
+            {
+                List<String> redirects = getRedirects(url);
+                if (redirects.isEmpty())
+                {
+                    result.add(url);
+                    continue;
+                }
+
+                String redirect = redirects.get(redirects.size() - 1);
```
Comment (valfirst) [inline vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java:85]:
```suggestion
                String targetLocation = redirects.get(redirects.size() - 1);
```

File: vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java [hunk 3]
```diff
@@ -66,35 +68,55 @@ protected Set<String> filterResults(Stream<String> urls)
...
+                if (!allUrls.contains(redirect))
+                {
+                    result.add(url);
+                }
```
Comment (valfirst) [inline vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java:89]: what's the logic behind it?

File: vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java [hunk 4]
```diff
@@ -66,35 +68,55 @@ protected Set<String> filterResults(Stream<String> urls)
...
+
+                redirectChains.add(url + REDIRECT_JOINER + String.join(REDIRECT_JOINER, redirects));
```
Comment (valfirst) [inline vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java:91]: the string concatenation should happen at the logging stage, collect all the data into some data structure, e.g. Map

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java (hunk 3 above)
Comment (uarlouski) [reply vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java:89]: the logic is fully preserved with the previous version of this code

File: vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java (hunk 1 above)
Comment (uarlouski) [reply vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/transformer/AbstractFetchingUrlsTableTransformer.java]: reverted

**Review by valfirst — APPROVED** [review]

**Pull request merged**